model_compression_toolkit/core/common/mixed_precision/configurable_quantizer_utils.py

```diff
@@ -20,29 +20,44 @@
     CandidateNodeQuantizationConfig
 
 
-def verify_candidates_descending_order(node_q_cfg: List[CandidateNodeQuantizationConfig]):
+def verify_candidates_descending_order(node_q_cfg: List[CandidateNodeQuantizationConfig],
+                                       kernel_attr: str = None):
     """
     Make sure the candidates configurations arrives in descending order.
 
     Args:
         node_q_cfg: Quantization configuration candidates of the node that generated the layer that will
                     use this quantizer.
+        kernel_attr: Kernel attribute name to verify its candidates order, if this is a weights configurable node.
 
-    Returns:
+    Returns: True if the candidates are ordered in descending order for the kernel attribute bit-width first
+    (if this is a weights configurable node) and activation bit-width as a secondary order.
 
     """
-    curmax = (np.inf, np.inf)
-    n_candidate_bits = [(x.weights_quantization_cfg.weights_n_bits, x.activation_quantization_cfg.activation_n_bits)
-                        for x in node_q_cfg]
-    for candidate_bits in n_candidate_bits:
-        assert candidate_bits < curmax, f"Node's quantization configuration candidates should arrive in " \
-                                        f"descending order of (weights_nbits, activation_nbits)."
-        curmax = candidate_bits
+
+    if kernel_attr is not None:
+        curmax = (np.inf, np.inf)
+        n_candidate_bits = [(x.weights_quantization_cfg.get_attr_config(kernel_attr).weights_n_bits,
+                             x.activation_quantization_cfg.activation_n_bits)
+                            for x in node_q_cfg]
+        for candidate_bits in n_candidate_bits:
+            assert candidate_bits < curmax, f"Node's quantization configuration candidates should arrive in " \
+                                            f"descending order of (weights_nbits, activation_nbits)."
+            curmax = candidate_bits
+    else:
+        # The candidates are only activation configurable
+        curmax = np.inf
+        n_candidate_bits = [x.activation_quantization_cfg.activation_n_bits for x in node_q_cfg]
+        for candidate_bits in n_candidate_bits:
+            assert candidate_bits < curmax, f"Node's quantization configuration candidates should arrive in " \
```

Comment:
Logger?

Reply:
It's more of an inner validation that the user doesn't have anything to do about (he shouldn't even know about that), so I'll keep it just an assertion for now (it was an assertion before it was just copied from somewhere else so it seems new code but it's not)